### ISSUE TITLE
Fails to build in a directory with spaces in the name

### ISSUE THREAD

**Issue by michaliskambi** [kcat/openal-soft#889]
CMakeLists.txt lines

```
set_property(TARGET ${IMPL_TARGET} APPEND_STRING PROPERTY
            LINK_FLAGS " -Wl,--version-script=${OpenAL_SOURCE_DIR}/libopenal.version")
```

cause problems when building in a directory with spaces in the name. 

We workarounded it in https://github.com/castle-engine/castle-engine/commit/063d33c95ce8a01cf26940d4b2e7b688731d19c1 by adding apostrophes around the argument, making it 

```
set_property(TARGET ${IMPL_TARGET} APPEND_STRING PROPERTY
            LINK_FLAGS " '-Wl,--version-script=${OpenAL_SOURCE_DIR}/libopenal.version'")
```

Admittedly my knowledge of cmake is too limited to judge whether this is a good solution, but at least it makes building work on Linux when the target SO is inside Android project which is inside a directory with spaces.

**Comment by kcat** [comment]
Commit 4227cb83ced45b7a25e3720fe9d75381a40fe0ab changes it to use `set_target_options` instead of `set_property`, which should hopefully properly escape spaces and special characters. Are you able to check if this works for you?

**Comment by michaliskambi** [comment]
Thank you --- I tested the solution from https://github.com/kcat/openal-soft/commit/4227cb83ced45b7a25e3720fe9d75381a40fe0ab and yes, it works nicely for me. The build goes OK even when project is in directory with spaces.

Note: This change does require CMake version >= 3.13. Trying to use it with outdated CMake version (as it happened, I was initially using CMake 3.10.2 distributed with old Android SDK) results in an error:

```
Unknown CMake command "target_link_options".
```

Searching, indeed `target_link_options` is only in CMake >= 3.13: 
https://stackoverflow.com/questions/25043458/does-cmake-have-something-like-target-link-options/53190881#53190881

I would recommend to change the `cmake_minimum_required` at the top to `cmake_minimum_required(VERSION 3.13)`, to signal this requirement explicitly. Right now it says `cmake_minimum_required(VERSION 3.0.2)` ( https://github.com/kcat/openal-soft/blob/master/CMakeLists.txt#L3 ) which is then no longer true.

Once I upgraded everything and I was using the latest CMake (3.22.1) from Android SDK, everything works flawlessly.
